pkg/instance/instance.go

```diff
@@ -1088,21 +1088,22 @@ func (i *Instance) WaitInstanceIsRunning(ctx context.Context) error {
 	if !i.IsInState(Started) {
 		return ErrWaitingForInstanceNotAllowed.WithParams(i.state.String())
 	}
-	timeout := time.After(1 * time.Minute)
 	tick := time.NewTicker(1 * time.Second)
 
 	for {
+		running, err := i.IsRunning(ctx)
```

Comment:
Since the ticker waits for the defined duration even for the very first trigger, in this example it waits one second before checking if the instance is running or not. Moving the check above the ticker solves it. Although in practice, I am not sure if that helps as usually a pod takes more than a second to be running.

The other change is to use the incoming context instead of having a local timeout.

Reply:
this makes sense, from a code structure standpoint we should always check or perform an action in a loop and then wait. Since in the future things might speed up and sometimes be satisfied on the first loop iteration.